distributed/tests/test_worker.py

```diff
@@ -2589,81 +2589,6 @@ def __reduce__(self):
     assert "return lambda: 1 / 0, ()" in logvalue
 
 
-@gen_cluster(client=True)
-async def test_gather_dep_exception_one_task(c, s, a, b):
-    """Ensure an exception in a single task does not tear down an entire batch of gather_dep
```

Comment:
This is misleading.
This test was testing resilience to an exception in the *transitions* of a single task after gather_dep - which should be dealt with through `@fail_hard`.

A _legitimate_ exception in a single key of the bundle in gather_dep, namely a task that fails to unpickle, does make the whole gather_dep fail for all tasks. There's no code whatsoever to deal with this use case.